climpred/relative_entropy.py

```diff
@@ -0,0 +1,343 @@
+        # print('transpose(*dims)',*dims,'time should be first')
+        base_to_calc_eofs = control.stack(
+            new=tuple(non_spatial_control_dims)).rename({'new': 'time'}).set_index({'time': 'time'}).transpose(*dims)
+    elif 'initialization' in control.dims and 'member' not in control.dims:  # PM_control
+        base_to_calc_eofs = control.rename({'initialization': 'time'})
+    else:
+        raise ValueError('adapt you inputs to PM- or DPLE/LENS-style')
+
+    # print('calc EOF base',base_to_calc_eofs.dims,'time should be first')
+    solver = Eof(base_to_calc_eofs, weights=wgts)
+
+    re_leadtime_list = []
+    lead_times = initialized.time.values[:ntime]
+    initializations = initialized.initialization.values
+    # DoTo: parallelize this double loop
+    for init in initializations:  # loop over initializations
+        rl, sl, dl = ([] for _ in range(3))  # lists to store results in
+        for t in lead_times:  # loop over lead time
+            # P_b base distribution
+            pc_b = solver.projectField(anom_b.sel(initialization=init, time=t)
+                                             .drop('time')
+                                             .rename({'member': 'time'}),
+                                       neofs=neofs, eofscaling=0,
+                                       weighted=False)
+
+            mu_b = pc_b.mean('time')
+            sigma_b = xr.DataArray(np.cov(pc_b.T))
+
+            # P_x initialization distribution
+            pc_x = solver.projectField(anom_x.sel(initialization=init, time=t)
+                                             .drop('time')
+                                             .rename({'member': 'time'}),
+                                       neofs=neofs, eofscaling=0,
+                                       weighted=False)
+
+            mu_x = pc_x.mean('time')
+            sigma_x = xr.DataArray(np.cov(pc_x.T))
+
+            r, d, s = _relative_entropy_formula(sigma_b, sigma_x, mu_x, mu_b,
+                                                neofs)
+
+            rl.append(r)
+            sl.append(s)
+            dl.append(d)
+
+        re_leadtime_list.append(xr.Dataset({'R': ('time', rl),
+                                            'S': ('time', sl),
+                                            'D': ('time', dl)}))
+
+    re = xr.concat(re_leadtime_list, dim='initialization').assign(
+        initialization=initializations, time=lead_times)
+
+    return re
+
+
+def _shuffle(ds, dim='initialization'):
+    """Shuffle ensemble members to uninitialize the data."""
+    old_dim_range = ds[dim]
+    shuffled = ds.sel({dim: np.random.permutation(ds[dim])})
+    if isinstance(ds, xr.DataArray):
+        shuffled[dim] = old_dim_range
+    elif isinstance(ds, xr.Dataset):
+        shuffled = shuffled.assign({dim: old_dim_range})
+    shuffled = shuffled.sortby(dim)
+    return shuffled
+
+
+def create_uninitialized_ensemble_from_control(ds, control, member_label=[1, 2, 3, 4, 5]):
```

Comment:
Similar to earlier, why is `[1, 2, 3, 4, 5]` the default for `member_label`? Can this just be a required input or done in the body of the function? It seems arbitrary to decide on 5 members here.